recipes/imagemagick/all/conanfile.py

```diff
@@ -61,375 +75,321 @@ class ImageMagicConan(ConanFile):
+        "with_flif": False,
+        "with_fpx": False,
+        "with_gslib": False,
+        "with_gvc": False,
+        "with_lqr": False,
+        "with_raqm": False,
+        "with_wmf": False,
         "utilities": True,
     }
     exports_sources = "patches/*"
-
+    win_bash = True
     _autotools = None
 
     @property
     def _source_subfolder(self):
-        return (
-            "ImageMagick"  # name is important, VisualMagick uses relative paths to it
-        )
+        return "source_subfolder"
 
     @property
     def _build_subfolder(self):
         return "build_subfolder"
 
-    @property
-    def _is_msvc(self):
-        return self.settings.compiler == "Visual Studio"
+    def layout(self):
+        self.folders.source = self._source_subfolder
+        self.folders.build = self._build_subfolder
 
     @property
     def _modules(self):
         return ["Magick++", "MagickWand", "MagickCore"]
 
     def validate(self):
-        if self.settings.os == "Windows":
+        if self.settings.os == "Windows" and scm.Version(self.version) < "7.1.0-45":
             raise ConanInvalidConfiguration(
-                "Windows builds of ImageMagick require MFC which cannot currently be sourced from CCI."
+                "This version of ImageMagick can't be built on Windows!"
             )
 
+        # TODO
+        unsupported_libs = {
+            "raw": self.options.with_raw,
+            "jxl": self.options.with_jxl,
+            "djvu": self.options.with_djvu,
+            "rsvg": self.options.with_rsvg,
+            "autotrace": self.options.with_autotrace,
+            "dps": self.options.with_dps,
+            "flif": self.options.with_flif,
+            "fpx": self.options.with_fpx,
+            "gslib": self.options.with_gslib,
+            "gvc": self.options.with_gvc,
+            "lqr": self.options.with_lqr,
+            "raqm": self.options.with_raqm,
+            "wmf": self.options.with_wmf
+        }
+
+        for lib, is_required in unsupported_libs.items():
+            if is_required:
+                raise ConanInvalidConfiguration(
+                    f"Building ImageMagick with '{lib}' package is not supported (yet)."
+                )
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if microsoft.is_msvc(self):
+            self.options.with_openmp = False
+            # FIXME: openexr causes link errors on windows
+            self.options.with_openexr = False
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def build_requirements(self):
+        if hasattr(self, "settings_build"):
+            self.build_requires("automake/1.16.5")
+        # Building on windows is currently only supported using the msys2 subsystem
+        if self._settings_build.os == "Windows":
```

Comment:
```suggestion
        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```
Keep installed msys2 as a possibility